### PR TITLE
[prim] Split out secded primitives

### PR DESCRIPTION
**Pull request by tomroberts-lowrisc**
- Add some new secded primitives required by the Ibex icache
- Use a separate core file for secded primitives which can be included as
  a dependency in Ibex and OpenTitan
- Relates to #1231

Signed-off-by: Tom Roberts <tomroberts@lowrisc.org>

### CODE REVIEW

File: hw/ip/prim/prim.core [hunk 1]
```diff
@@ -15,6 +15,7 @@ filesets:
       - lowrisc:prim:pad_wrapper
       - lowrisc:prim:prim_pkg
       - lowrisc:prim:clock_mux2
+      - lowrisc:prim:secded
```
Comment (imphil) [inline hw/ip/prim/prim.core:18]:
```suggestion
      - lowrisc:prim:secded # for prim_ram_2p_adv
```

(And potentially push this one up to the top of the list.)

Comment (tomroberts-lowrisc) [reply hw/ip/prim/prim.core]: Done in latest patchset

**Comment by eunchan** [comment]
Hope SECDED gen has no bug.. I should prepare FPV assertions to check.

**Comment by tjaychen** [comment]
should we add a small util with an hjson of all the secded primitives that
have been generated? That way if our generation ever changes in the future,
we can just run a script to re-generate all the files we need (and not
accidentally forget one).

**Comment by eunchan** [comment]
Unfortunately, current secded gen uses random seed to generate parity 
matrix. we should choose fixed seed to get the same result with multiple 
run.

**Comment by tjaychen** [comment]
ooo that's right. I forgot about that, this is the weight search right?
I actually don't think it's necessary that we always generate the same..I
meant more that if we find a bug, and need to re-generate all of them, we
don't accidentally forget one.

**Pull request merged**